flexget/plugins/input/gazelle.py

```diff
@@ -0,0 +1,461 @@
+        if not no_login and r.is_redirect and r.next.url == "{}/login.php".format(self.base_url):
+            log.warning("Redirected to login page, reauthenticating and trying again")
+            self.authenticate(force=True)
+            return self.request(no_login=True, **params)
+
+        if r.status_code != 200:
+            raise PluginError("{} returned a non-200 status code".format(self.base_url))
+
+        try:
+            json_response = r.json()
+            if json_response['status'] != "success":
+
+                # Try to deal with errors returned by the API
+                error = json_response.get('error', json_response.get('status'))
+                if not error or error == "failure":
+                    error = json_response.get('response', str(json_response))
+
+                raise PluginError("{} gave a failure response of '{}'".format(self.base_url, error))
+            return json_response['response']
+        except (ValueError, TypeError, KeyError):
+            raise PluginError("{} returned an invalid response".format(self.base_url))
+
+    def search_results(self, params):
+        """Generator that yields search results"""
+        page = 1
+        pages = None
+        while page <= self.max_pages:
+            if pages and page >= pages:
+                break
+
+            log.debug("Attempting to get page %d of search results", page)
+            result = self.request(action='browse', page=page, **params)
+            if not result['results']:
+                break
+            for x in result['results']:
+                yield x
+
+            pages = result.get('pages', pages)
+            page += 1
+
+        if page > self.max_pages:
+            log.warning("Stopped after %d pages (out of %d total pages)",
+                        self.max_pages, pages)
+
+    def get_entries(self, search_results):
+        """Generator that yields Entry objects from search results"""
+        for result in search_results:
+            # Get basic information on the release
+            info = dict((k, result[k]) for k in ('groupId', 'groupName'))
+
+            # Releases can have multiple download options
+            for tor in result['torrents']:
+                temp = info.copy()
+                temp.update(dict((k, tor[k])
+                            for k in ('torrentId',)))
+
+                yield Entry(
+                    title="{groupName} ({groupId} - {torrentId}).torrent".format(**temp),
+                    url="{}/torrents.php?action=download&id={}&authkey={}&torrent_pass={}"
+                        "".format(self.base_url, temp['torrentId'], self.authkey, self.passkey),
+                    torrent_seeds=tor['seeders'],
+                    torrent_leeches=tor['leechers'],
+                    # Size is returned in bytes, convert to MB for compat with the content_size plugin
+                    content_size=tor['size'] / (1024 ** 2)
```

Comment:
We have a util for this

Reply:
I did see that but didn't think it was appropriate. That function takes a string, guesses the correct units, and converts it to bytes. To use it I would have to do something like:
```python
Entry(
    content_size=parse_filesize(str(tor['size']) + "b")
)
```
Is that preferred over just converting it manually?

Reply:
It might be a good idea to create a function that does that. It's not the first time we've needed it.